miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultInstrumentService.java

```diff
@@ -58,25 +54,43 @@ public Instrument getByUpgradedInstrumentId(long instrumentId) throws IOExceptio
   }
 
   @Override
-  public Long create(Instrument instrument) throws IOException {
+  public long create(Instrument instrument) throws IOException {
     authorizationManager.throwIfNonAdmin();
     loadChildEntities(instrument);
+    validateChange(instrument, null);
     return save(instrument);
   }
 
   @Override
-  public void update(Instrument instrument) throws IOException {
+  public long update(Instrument instrument) throws IOException {
     authorizationManager.throwIfNonAdmin();
     Instrument managed = get(instrument.getId());
+    validateChange(instrument, managed);
     applyChanges(managed, instrument);
     loadChildEntities(managed);
-    save(managed);
+    return save(managed);
   }
 
   private long save(Instrument instrument) throws IOException {
     return instrumentDao.save(instrument);
   }
 
+  private void validateChange(Instrument instrument, Instrument beforeChange) throws IOException {
+    List<ValidationError> errors = new ArrayList<>();
+
+    if (ValidationUtils.isSetAndChanged(Instrument::getName, instrument, beforeChange) && getByName(instrument.getName()) != null) {
+      errors.add(new ValidationError("name", "There is already an instrument with this name"));
+    }
+    if (ValidationUtils.isSetAndChanged(Instrument::getUpgradedInstrument, instrument, beforeChange)
+        && getByUpgradedInstrumentId(instrument.getUpgradedInstrument().getId()) != null) {
```

Comment:
This line throws a NPE when creating a new instrument

Reply:
Fixed. `ValidationUtils#isSetAndChanged` was returning true when the field was _not_ set if `beforeChange` was null (on create)